### PR TITLE
OAS: rework data feeding + sampleIDs

### PR DESCRIPTION
**Pull request by undera**
1. Change it into reading directly from DB, since we need entryIDs
2. Add sample IDs on different levels

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -18,6 +18,7 @@ jobs:
   run-unit-tests:
     name: Unit Tests
     runs-on: ubuntu-latest
+    timeout-minutes: 30 # TODO: is it low enough?
```
Comment (IgorGov) [inline .github/workflows/test.yml:21]: did you reached timeout with this workflow?

Comment (IgorGov) [inline .github/workflows/test.yml:21]: did you reached timeout with this workflow?

Comment (IgorGov) [reply .github/workflows/test.yml:21]: i don't think it should be more than 10 minutes.
if it more then that we need to rethink how we running the tests

Comment (IgorGov) [reply .github/workflows/test.yml:21]: i don't think it should be more than 10 minutes.
if it more then that we need to rethink how we running the tests

**Review by IgorGov — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (undera) [reply .github/workflows/test.yml:21]: I have observed once that the job were taking very long time. I thought that something in unit tests made them hang indefinitely. IMO having safety valve timeout is a good idea. I will change it into 10m

**Review by IgorGov — APPROVED** [review]

**Pull request merged**